pkg/auth/providerrefresh/refresher.go

```diff
@@ -159,33 +167,56 @@ func (r *refresher) refreshAttributes(attribs *v3.UserAttribute) (*v3.UserAttrib
 		return nil, err
 	}
 
-	loginTokens = make(map[string][]*v3.Token)
-	derivedTokens = make(map[string][]*v3.Token)
+	loginTokens = make(map[string][]accessor.TokenAccessor)
+	derivedTokens = make(map[string][]accessor.TokenAccessor)
+
+	// List all tokens, v3 and ext.
+	// For ext tokens we actually filter for the user here.
+
+	allV3Tokens, err := r.tokenLister.List("", labels.Everything())
+	if err != nil {
+		return nil, err
+	}
 
-	allTokens, err := r.tokenLister.List("", labels.Everything())
+	allExtTokens, err := r.extTokenStore.ListForUser(user.Name)
 	if err != nil {
 		return nil, err
 	}
 
+	// Merge the separate lists into a unified set
+	allTokens := make([]accessor.TokenAccessor, 0, len(allV3Tokens)+len(allExtTokens.Items))
+	for _, token := range allV3Tokens {
+		allTokens = append(allTokens, token)
```

Comment:
You could simplify this to `allTokens = append(allTokens, allV3Tokens...)`.

Reply:
Actually no. The allV3Tokens is `[]v3.Token` and you cannot append such a slice to a `[]Accessor (interface)`.
It only works when appending as individual elements. IOW in a loop :(